config/reader.go

```diff
@@ -284,12 +419,18 @@ func Read(
 	filePath string,
 	logger golog.Logger,
 ) (*Config, error) {
-	buf, err := envsubst.ReadFile(filePath)
```

Comment:
removing this something that I had to do because it will try and map ${} to BASH variables which isn't set and so it replaces the placeholders with empty strings. Trying to find a way of doing this better because replacing env vars is something we will definitely want to do.

Reply:
There is really no good solution to this, but i think we can probably replace the env vars easily with our version of this and using GetEnv() functionality. Will try and add something.

Reply:
not actually sure if this is used, i just added an env var in a fake config and tried to run stable RDK and it broke it. 

I think that if this is the case, env var replacement could become an extension of this code?

Reply:
I had no idea we did this currently. I think we should keep it because I'm not sure how/where it is used and don't want to break anyone.

How about instead of implementing a custom replacer, what if we moved the env var substitution to after the placeholder one? Then the RDK will replace packages first, then env vars afterwards. Anything that doesn't resolve overall will be an empty string.

I think that just means moving this to after the places where package paths are resolved and then using envsubst.Bytes instead of reading from file.

Reply:
i agree with this ^ but envsubst replaces only ${VAR} so checking on ${env.var} would not be backwards compatible. Could possibly do both for env replacing if we want to support this in the future but this is only for backward compatibility for now.